v2/observer/jobs.go

```diff
@@ -46,6 +47,16 @@ func (o *observer) syncJobPods(ctx context.Context) {
 func (o *observer) syncJobPod(obj interface{}) {
 	pod := obj.(*corev1.Pod)
 
+	// Fetch the cancel func associated with a timed job pod
+	lookup := pod.Namespace + "/" + pod.Name
+	cancelTimer, exists := o.timedPodsSet[lookup]
```

Comment:
Nit. If you _were_ to successfully find a `cancelTimer` in the map here, you don't ever end up using it, so the code is slightly easier to follow here if you use `_, exists := o.timedPodsSet[lookup]` and only define `cancelTimer` within the conditional below if/when it is necessary.

Reply:
I think we also need to protect this map with a mutex.

The inner workings of the k8s lister/watcher is somewhat opaque to me, but I am at least questioning the possibility that the `syncJobPod()` function can be invoked more than once, concurrently, on the _same_ pod. For instance, syncing one given pod's state n _and_ state n + 1 concurrently.

(Or maybe the lister/watcher is built to avoid that condition, otherwise, there would be race conditions all of the place in k8s, right? Bottom line is I simply don't know. Perhaps we need to research this.)

At any rate... the reason I bring it up is because their _may_ be a need to protect this map with a mutex so that we never accidentally start multiple timer loops for the same pod.

Reply:
After a convo with @thomastaylor312, I withdraw any concern that a mutex may be needed to protect concurrent access to this map.